onnxruntime/core/providers/cuda/cuda_execution_provider.h

```diff
@@ -23,7 +23,7 @@ struct CUDAExecutionProviderInfo {
 // Logical device representation.
 class CUDAExecutionProvider : public IExecutionProvider {
  public:
-  explicit CUDAExecutionProvider(const CUDAExecutionProviderInfo& info);
+  explicit CUDAExecutionProvider(const CUDAExecutionProviderInfo& info, size_t cuda_mem_limit = std::numeric_limits<size_t>::max());
```

Comment:
Since CUDAExecutionProviderInfo is specifically meant for constructing CUDAExecutionProvider, cuda_mem_limit should be part of the info struct. What's the reason for it to be a separate param?